include/pybind11/class_support.h

```diff
@@ -27,6 +27,10 @@ extern "C" inline int pybind11_static_set(PyObject *self, PyObject *obj, PyObjec
     return PyProperty_Type.tp_descr_set(self, cls, value);
 }
 
+static PyTypeObject *type_with_ref_incd(PyTypeObject *type) {
+    return (PyTypeObject *)(handle((PyObject *)(type)).inc_ref().ptr());
```

Comment:
Actually, scracth that: it's much simpler to just bypass the pybind helper classes completely here:
```C++
    Py_INCREF(type);
    return type;
```